### PR TITLE
Add troubleshooting doc for SQL Server integration

### PR DESCRIPTION
**Pull request by jmeunier28**
### What does this PR do?

Prior to this PR, there were public docs around troubleshooting for the Postgres & MySQL integrations, but nothing for SQL Server. This adds docs around known issues that we see come up in support (mostly diagnosing connection issues / issues with the driver)

### Motivation

In an effort to decrease the number of support requests we get around the SQL Server integration setup, we are adding better troubleshooting docs to help customers diagnose issues on their own 

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+If you are getting a login error connecting, it is important to verify you can log into the instance as the datadog agent user first. An easy way to do this is via a cmd line utility such as `sqlcmd`.
+
+For example:
+
+```bash
+# this example uses SQL Authentication
+sqlcmd -S <INSTANCE_ENDPOINT> -U datadog -P <DATADOG_PASSWORD> -d master
+
+# this examples uses Windows Authentication
```
Comment (EmmaOnThursday) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:37]: typo: `this example`

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+If you are using the latest version of the Microsoft OLE DB Driver for SQL Server, and trying to connect to a SQL Server instance which requires encrypted connections, you can use the following workarounds:
+
+1. If you are using a self-signed certificate and the Force Encryption setting on the server (`rds.force_ssl=1` on AWS) to ensure clients connect with encryption:
+
+   - Change to a certificate that is trusted as part of the client's trust chain
+   - Add the self-signed certificate as a trusted certificate on the client
+   - Add `TrustServerCertificate=yes;` to the connection string
+
+This is described in more detail [in the microsoft documentation][7]
```
Comment (EmmaOnThursday) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:95]: nit: capitalize Microsoft

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+#### SQL Server user tag is missing on the Query Metrics page
+
+The `user` tag has been removed from all sqlserver metric telemetry.
+
+The `user` tag was previously incorrect and did not reflect the user who executed the query in the Query Metrics UI. This is due to a technical limitation in the tables that we are querying in sqlserver for this data.
+
+#### Why are there so many “CREATE PROCEDURE” queries?
+
+In versions of the agent older than 7.40.0, there exists a bug where `PROCEDURE` statistics are over counted. This leads to seeing many executions of `CREATE PROCEDURE...` in the database-monitoring Query Metrics UI. In order to fix this issue, please upgrade to the latest version of the sqlserver agent.
```
Comment (EmmaOnThursday) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:165]: probably: `... please upgrade to the latest version of the Datadog agent` at the end

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+Should be set as:
+```
+host: sqlserver-foo.cfxxae8cilce.us-east-1.rds.amazonaws.com,1433
+```
+
+### Other common questions
+
+#### SQL Server user tag is missing on the Query Metrics page
+
+The `user` tag has been removed from all sqlserver metric telemetry.
```
Comment (EmmaOnThursday) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:159]: nit: SQL Server not sqlserver

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+The `user` tag was previously incorrect and did not reflect the user who executed the query in the Query Metrics UI. This is due to a technical limitation in the tables that we are querying in sqlserver for this data.
```
Comment (EmmaOnThursday) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:161]: nit: nit: SQL Server not sqlserver

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+`-2147217843` — **“login failed for user”**: this means the agent succeeded in establishing a connection to the host but the login was rejected for some reason.
+
+In order to troubleshoot this you can:
+
+1. Check the agent’s login credentials
+
+2. Try to login with those credentials manually using sqlcmd. For example: `sqlcmd -S localhost -U datadog -P ${SQL_PASSWORD} -d master`
+
+`-2147467259` — **“could not open database requested for login”**: this error appears either due to network issues or due to an unknown database. To troubleshoot:
```
Comment (EmmaOnThursday) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:73]: nit: last words `To troubleshoot:` unnecessary/duplicate following line

**Comment by jmeunier28** [comment]
Ty for review @EmmaOnThursday 🙇 I think I got it all

**Review by EmmaOnThursday — APPROVED** [review]
👍 this looks great

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+If the `datadog` user is unable to log into the SQL Server instance, please ensure the user has been created and given the proper permissions according the [setup documentation][1].
+
+Microsoft also provides a helpful doc on troubleshooting these types of errors, which can be [followed here][5].
+
+### SQL Server Unable to connect due to “Invalid connection string attribute”
+
+The following ADO Providers are supported on Windows: `SQLOLEDB`, `MSOLEDBSQL`, `MSOLEDBSQL19`, `SQLNCLI11`.
+
+The `SQLOLEDB` and `SQLNCLI11` providers produce a particularly uninformative error message, `Invalid connection string attribute`. For example:
```
Comment (justiniso) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:48]:
```suggestion
The `SQLOLEDB` and `SQLNCLI11` providers produce the error message `Invalid connection string attribute` for a wide variety of possible issues. For example:
```

Comment (justiniso) [reply content/en/database_monitoring/setup_sql_server/troubleshooting.md:48]: But docs team, please feel free to suggest other language.

**Comment by cswatt** [comment]
created DOCS-4287 to track

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+        driver: ODBC Driver 13 for SQL Server
+        username: <USERNAME>
+        password: <PASSWORD>
+    ```
+
+## Other common questions
+
+### SQL Server user tag is missing on the Query Metrics page
+
+The `user` tag has been removed from all SQL Server metric telemetry.
```
Comment (justiniso) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:202]:
```suggestion
The `user` tag is no longer supported for Query Metrics due to technical limitations in SQL Server which prevent the collection of the correct user running queries.

The `user` tag is available for Query Activity events and Database Load metrics.
```

File: content/en/database_monitoring/setup_sql_server/troubleshooting.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+
+The `user` tag was previously incorrect and did not reflect the user who executed the query in the Query Metrics UI. This is due to a technical limitation in the tables that we are querying in SQL Server for this data.
```
Comment (justiniso) [inline content/en/database_monitoring/setup_sql_server/troubleshooting.md:204]:
```suggestion

```

**Review by justiniso — CHANGES_REQUESTED** [review]

**Review by jtappa — CHANGES_REQUESTED** [review]
this page also needs to be added to the left navigation in menus.en.yaml, you can use the other troubleshooting pages in the DBM section as a reference, but let me know if you need help with this!

**Comment by jmeunier28** [comment]
@jtappa this look right? 40523a304791a709e69d250174de9a5bbceb7b74

**Comment by cswatt** [comment]
@jmeunier28 nav looks good!

**Review by cswatt — APPROVED** [review]

**Pull request merged**